CMakeLists.txt

```diff
@@ -70,6 +70,11 @@ IF(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
   ADD_COMPILE_OPTIONS("-Wno-psabi")
 ENDIF()
 
+# Xcode does not support COMPILE_FLAGS according to https://cmake.org/cmake/help/latest/prop_sf/COMPILE_FLAGS.html
+IF(CMAKE_GENERATOR STREQUAL Xcode)
+  set(XCODE ON)
+ENDIF()
+
```

Comment:
Don't define `XCODE` variable, it may lead to name collisions. Just use `IF(CMAKE_GENERATOR STREQUAL Xcode)` instead of of `IF(XCODE)`

Reply:
Refactored